cmd/zetae2e/local/bitcoin.go

```diff
@@ -5,71 +5,147 @@ import (
 	verbose bool,
-	initBitcoinNetwork bool,
-	testNames ...string,
-) func() error {
-	return func() (err error) {
-		account := conf.AdditionalAccounts.UserBitcoin
-		// initialize runner for bitcoin test
-		bitcoinRunner, err := initTestRunner(
-			"bitcoin",
-			conf,
-			deployerRunner,
-			account,
-			runner.NewLogger(verbose, color.FgYellow, "bitcoin"),
-		)
-		if err != nil {
-			return err
-		}
+	initNetwork bool,
+	depositTests []string,
+	withdrawTests []string,
+) {
+	// initialize runner for deposit tests
+	runnerDeposit := initRunnerDeposit(conf, deployerRunner, verbose, initNetwork)
 
-		bitcoinRunner.Logger.Print("🏃 starting Bitcoin tests")
-		startTime := time.Now()
+	// initialize runner for withdraw tests
+	runnerWithdraw := initRunnerWithdraw(conf, deployerRunner, verbose, initNetwork)
+
+	// initialize funds
+	// send BTC to TSS for gas fees and to tester ZEVM address
+	if initNetwork {
+		// mine 101 blocks to ensure the BTC rewards are spendable
+		// Note: the block rewards can be sent to any address in here
+		_, err := runnerDeposit.GenerateToAddressIfLocalBitcoin(101, runnerDeposit.BTCDeployerAddress)
+		require.NoError(runnerDeposit, err)
+
+		// send BTC to ZEVM addresses
+		runnerDeposit.DepositBTC(runnerDeposit.EVMAddress())
+		runnerDeposit.DepositBTC(runnerWithdraw.EVMAddress())
+	}
+
+	// create test routines
+	routineDeposit := createTestRoutine(runnerDeposit, depositTests)
+	routineWithdraw := createTestRoutine(runnerWithdraw, withdrawTests)
 
-		// funding the account
+	// start test routines
+	eg.Go(routineDeposit)
+	eg.Go(routineWithdraw)
+}
+
+// initRunnerDeposit initializes the runner for deposit tests
+func initRunnerDeposit(
+	conf config.Config,
+	deployerRunner *runner.E2ERunner,
+	verbose, initNetwork bool,
+) *runner.E2ERunner {
+	var (
+		name         = "btc_deposit"
+		account      = conf.AdditionalAccounts.UserBitcoin1
+		createWallet = true // deposit tests need Bitcoin node wallet to handle UTXOs
+	)
+
+	return initRunner(name, account, conf, deployerRunner, verbose, initNetwork, createWallet)
+}
+
+// initRunnerWithdraw initializes the runner for withdraw tests
+func initRunnerWithdraw(
+	conf config.Config,
+	deployerRunner *runner.E2ERunner,
+	verbose, initNetwork bool,
+) *runner.E2ERunner {
+	var (
+		name         = "btc_withdraw"
+		account      = conf.AdditionalAccounts.UserBitcoin2
+		createWallet = false // withdraw tests DON'T use Bitcoin node wallet
+	)
+
+	return initRunner(name, account, conf, deployerRunner, verbose, initNetwork, createWallet)
+}
+
+// initRunner initializes the runner for given test name and account
+func initRunner(
+	name string,
+	account config.Account,
+	conf config.Config,
+	deployerRunner *runner.E2ERunner,
+	verbose, initNetwork, createWallet bool,
+) *runner.E2ERunner {
+	// initialize runner for bitcoin test
+	runner, err := initTestRunner(
+		name,
+		conf,
+		deployerRunner,
+		account,
+		runner.NewLogger(verbose, color.FgYellow, name),
```

Comment:
Not strong opinion on this but maybe we could use a different shade of yellow for both runner

Reply:
can do